backend/Functions/Edna.Assignments/Edna.Assignments/AssignmentsApi.cs

```diff
@@ -112,19 +155,65 @@ private async Task<AssignmentEntity> FetchAssignment(CloudTable table, string as
         {
-            AssignmentEntity assignmentEntity = await FetchAssignment(table, assignmentId);
-            if (assignmentEntity == null)
-                return new NotFoundResult();
-
             assignmentEntity.PublishStatus = newPublishStatus.ToString();
             assignmentEntity.ETag = "*";
 
             await assignmentEntityCollector.AddAsync(assignmentEntity);
             await assignmentEntityCollector.FlushAsync();
 
+            _logger.LogInformation($"Changed Published status to {assignmentEntity.PublishStatus}");
+
+            return new OkResult();
+        }
+    
+        private async Task<IActionResult> ValidateUser(HttpRequest req, AssignmentDto assignmentDto, PlatformsClient platformsClient, INrpsClient nrpsClient, Lti1MembershipClient membershipClient)
+        {
+            if (!req.Headers.TryGetUserEmails(out List<string> userEmails))
+            {
+                _logger.LogError("Could not get user email.");
+                return new BadRequestErrorMessageResult("Bad Request: Could not get user email.");
+            }
+
+            if (userEmails.Count > 0)
+            {
+                if (assignmentDto.LtiVersion != LtiAdvantageVersionString)
+                {
+                    Membership userMembership = await membershipClient.GetMemberByEmail(assignmentDto.ContextMembershipsUrl, assignmentDto.OAuthConsumerKey, assignmentDto.ResourceLinkId, userEmails);
+                    if (userMembership == null)
+                    {
+                        _logger.LogError("no members");
+                        return new BadRequestErrorMessageResult("Bad Request: no members");
+                    }
+                    var role = userMembership.Role;
+                    if (role.Equals("Learner"))
+                    {
+                        _logger.LogError("Students cannot update an assignment");
+                        return new UnauthorizedResult();
+                    }
+                }
+                else
+                {
+                    Platform platform = await platformsClient.GetPlatform(assignmentDto.PlatformId);
+                    Member member = await nrpsClient.GetByEmail(platform.ClientId, platform.AccessTokenUrl, assignmentDto.ContextMembershipsUrl, userEmails);
+                    if (member == null)
+                    {
+                        _logger.LogError("no members");
+                        return new BadRequestErrorMessageResult("Bad Request: no members");
+                    }
+                    var roles = member.Roles;
+                    if (roles.Contains(Role.ContextLearner) || roles.Contains(Role.InstitutionLearner))
+                    {
+                        _logger.LogError("Students cannot update an assignment");
+                        return new UnauthorizedResult();
+                    }
+                }
+            }
+
+            _logger.LogInformation("Valid user");

```

Comment:
I don't believe we might need this, the errors and their corresponding logs should take care of it automatically. Extra logging might actually shadow the important information as the logs would become bulky.